lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -67,10 +71,64 @@ def destroy
     output = powershell(powershell_args, script_content)
     Puppet.debug "Destroy Dsc Resource returned: #{output}"
     data = JSON.parse(output)
+
     fail(data['errormessage']) if !data['errormessage'].empty?
-    true
+
+    initiate_reboot if data['rebootrequired'] == true
+
+    data
+  end
+
+  def initiate_reboot
+    Puppet.warning "A reboot is required to progress further. Notifying Puppet."
+    require 'pry';binding.pry;
+
+    catalog_resource = resource.catalog.resources.find_all do |r|
+      r.is_a?(Puppet::Type.type(:reboot))
+    end
+
+    require 'pry';binding.pry;
+    if catalog_resource.any?
+      return
+    end
+
+    target_resource = resolve_resource(catalog_resource[0])
+    require 'pry';binding.pry;
+
+    name = target_resource.name
+    type = target_resource.type
+    resource_key = [type, name].join('/')
+
+    if target_resource.respond_to?(:autonotify)
+      target_resource.autonotify
+    end
```

Comment:
This was a placeholder until I got the correct method to look for as well as call.